### PR TITLE
Starting Projects Dashboard for Platform Admin closes #182

### LINKED ISSUE #182: As a platform admin, I can go to the admin dashboard, click on Projects and get a listing of all projects.

### PR DESCRIPTION
**Pull request by mikedao**
Started the implementation of projects for platform admin. 
* Created link on main dashboard page.
* Created controller.
* Created View
* Wrote all necessary tests for this stage.

Will continue work with a smaller story and another branch tomorrow.

### CODE REVIEW

**Comment by EmilyMB** [comment]
@mikedao: just a general :question:  What was the thinking behind making a separate controller/view? Why not just use the tenant projects controller and views? We already have a separate projects controller for general users..

**Comment by mikedao** [comment]
Tenant projects controllers and views only show theirs. The thinking is that we have a master list. But here's the cool part - when we link to edit the page and delete projects we use the stuff we made in the tenants project controller, we dont have to rebuild that part of it, we just have to make a nice view to start

File: app/controllers/admin/projects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Admin::ProjectsController < Admin::BaseController
+  def index
+    @project = Project.new
```
Comment (EmilyMB) [inline app/controllers/admin/projects_controller.rb]: Why are we making a new project in the index view?

File: app/views/admin/base/dashboard.html.erb [hunk 2]
```diff
@@ -5,5 +5,6 @@
 
 <p><%= link_to "Categories", admin_categories_path %></p>
 <p><%= link_to "Tenants", admin_tenants_path %></p>
```
Comment (EmilyMB) [inline app/views/admin/base/dashboard.html.erb]: I thought we took this out already? I do prefer "borrowers" to "tenants". Personally, if these are the only three links, I don't see anything wrong with having them all in the navbar...

File: app/views/admin/projects/index.html.erb [hunk 3]
```diff
@@ -19,42 +19,20 @@
   </div>
 </div>
 
-<div class="row">
-  <% @projects.each do |project| %>
-  <div class="box col-lg-3">
```
Comment (EmilyMB) [inline app/views/admin/projects/index.html.erb]: Having it as a box class helps later with formatting, I think.

File: app/views/admin/projects/index.html.erb [hunk 4]
```diff
@@ -19,42 +19,20 @@
...
-      <%= f.submit 'Retire', class: "btn btn-default btn-update-quantity" %>
-      <% end %>
-    </div>
-  </th>
-  </thead>
-</table>
-    </div>
+    <% @projects.each do |project| %>
+      <tr>
+        <td><%= project.title %></td>
```
Comment (EmilyMB) [inline app/views/admin/projects/index.html.erb]: Are these going to link to something? And have the same edit button as in the tenant/projects/index? Or is that another PR to use a partial?

File: config/routes.rb [hunk 5]
```diff
@@ -16,6 +16,7 @@
     get "/dashboard", to: "base#dashboard"
     resources :categories
     resources :tenants, only: [:index, :edit, :update]
+    resources :projects, only: [:index, :edit]
```
Comment (EmilyMB) [inline config/routes.rb]: Nice use of limiting the routes :smile:

File: test/integration/admin_projects_test.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  test "an admin sees projects on the project dashboard" do
+    admin = create(:admin)
+    project = create(:project)
+    ApplicationController.any_instance.stubs(:current_user).returns(admin)
+
+    visit admin_projects_path
+
+    assert page.has_content?(project.title)
+  end
+end
```
Comment (EmilyMB) [inline test/integration/admin_projects_test.rb]: Off to a good start!

File: app/views/admin/projects/index.html.erb (hunk 4 above)
Comment (mikedao) [inline app/views/admin/projects/index.html.erb]: They are going to link to edit pages.

File: app/views/admin/projects/index.html.erb (hunk 3 above)
Comment (mikedao) [inline app/views/admin/projects/index.html.erb]: Bootstrap confused me.

File: app/controllers/admin/projects_controller.rb (hunk 1 above)
Comment (mikedao) [inline app/controllers/admin/projects_controller.rb]: Easier to have it there than another link. Also, it matches hwo we have it in category.

File: app/views/admin/base/dashboard.html.erb (hunk 2 above)
Comment (mikedao) [inline app/views/admin/base/dashboard.html.erb]: I think we have to have an orders as well.

**Comment by Jwan622** [comment]
So we are going to go with a having a master list of tenant projects (all of the projects) that a platform admin can see?

File: app/views/admin/projects/index.html.erb (hunk 4 above)
Comment (Jwan622) [inline app/views/admin/projects/index.html.erb]: Very neat. So the link for each project on this page goes to that project's edit page. What action will this link hit in which controller?

Comment (mikedao) [inline app/views/admin/projects/index.html.erb]: It will hit the tenants projects controller.

**Comment by mikedao** [comment]
@Jwan622 Yes.

**Comment by mikedao** [comment]
image

**Comment by Jwan622** [comment]
Looks good to me.

**Pull request merged**